.github/workflows/upstream-update.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Update with upstream repository
+on:
+    schedule:
+      - cron:  0 0 * * *
+      # scheduled at 00:00 every day
+    workflow_dispatch:
+
+jobs:
+    sync_with_upstream:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: Wandalen/wretry.action@v1.0.36
+              name: Checkout
+              with:
+                  action: actions/checkout@v3
+                  with: |
+                      token: ${{ secrets.WORKFLOW_TOKEN }}
+                      ref: master
```

Comment:
Should we run this against a test branch for a while first to see what happens ?

Reply:
I think our master will be a test branch. The development branch is our default so nothing should go wrong. The development will be our backup.